cpu/sam0_common/periph/i2c.c

```diff
@@ -363,6 +355,10 @@ static inline int _write(SercomI2cm *dev, const uint8_t *data, int length,
             return -EIO;
         }
     }
+    if (stop) {
+        /* Prepare stop command before write last byte */
```

Comment:
Is the comment correct? The last byte was already written.
Also, I get the feeling we should do something similar in _read. I mean, do the _stop after the while loop (if it is requested).

The comment probably belongs to the ACK/NACK, which you "prepare" it before doing the last read byte.